src/orbitalMechanics.cpp

```diff
@@ -0,0 +1,405 @@
+#include "mantella_bits/orbitalMechanics.hpp"
+
+// C++ standard library
+#include <cmath>
+#include <iostream>
+#include <stddef.h>
+#include <string>
+
+// Mantella
+#include "mantella_bits/assert.hpp"
+
+namespace mant {
+  namespace itd {
+
+    double degreeToRadian(
+        const double degree) {
+      return degree * (arma::datum::pi / 180.0);
+    }
+
+    std::vector<std::pair<arma::Col<double>::fixed<3>, arma::Col<double>::fixed<3>>> lambert(
+        const arma::Col<double>::fixed<3>& departurePosition,
+        const arma::Col<double>::fixed<3>& arrivalPosition,
+        const double transferTime,
+        const bool clockwiseDirection) {
+      verify(transferTime > 0.0, "lambert: transferTime must be greater than zero.");
+
+      // 1 - Getting lambda and Ts
+      double flightVectorLength = arma::norm(departurePosition - arrivalPosition);
+
+      double departurePositionVectorLength = arma::norm(departurePosition);
+      double arrivalPositionVectorLength = arma::norm(arrivalPosition);
+
+      double lengthSumWeight = (flightVectorLength + departurePositionVectorLength + arrivalPositionVectorLength) / 2.0;
+
+      arma::Col<double>::fixed<3> departurePositionNormalised = arma::normalise(departurePosition);
+      arma::Col<double>::fixed<3> arrivalPositionNormalised = arma::normalise(arrivalPosition);
+
+      arma::Col<double>::fixed<3> depatureArrivalCrossProduct = arma::normalise(arma::cross(departurePositionNormalised, arrivalPositionNormalised));
+
+      verify(depatureArrivalCrossProduct(2) != 0.0, "lambert: depatureArrivalCrossProduct must have a z component unequal zero.");
+
+      double lambdaSquare = 1.0 - flightVectorLength / lengthSumWeight;
+      lambda = std::sqrt(lambdaSquare);
+
+      arma::Col<double>::fixed<3> it1, it2;
+      if (depatureArrivalCrossProduct(2) < 0.0) {
+        lambda = -lambda;
+        it1 = arma::cross(departurePositionNormalised, depatureArrivalCrossProduct);
```

Comment:
Why are `it1` and `it2` even defined at this point, if they are not used prior to line 170?